src/core/backend.cc

```diff
@@ -189,4 +235,217 @@ InferenceBackend::Run(
   OnCompleteQueuedPayloads(status);
 }
 
+Status
+InferenceBackend::GenerateWarmupData(std::vector<WarmupData>* samples)
+{
+  static std::string warmup_data_folder = "warmup";
+
+  samples->clear();
+  for (const auto& warmup_setting : config_.model_warmup()) {
+    LOG_VERBOSE(1) << "Generating warmup sample data for '"
+                   << warmup_setting.name() << "'";
+
+    // Two passes. First pass to get max byte size for synthetic data
+    int64_t max_zero_byte_size = 0;
+    int64_t max_random_byte_size = 0;
+    for (const auto& input_meta : warmup_setting.inputs()) {
+      auto element_count = GetElementCount(input_meta.second.dims());
```

Comment:
There's a GetByteSize that takes element count and datatype... can you use that?

Reply:
I didn't use `GetByteSize()` since the element count will be needed for the STRING data type